### ISSUE TITLE
Fix retry logic in kcrypt-challenger

### ISSUE THREAD

**Issue by jimmykarily** [kairos-io/kairos#673]
Fix this: https://github.com/kairos-io/kcrypt-challenger/pull/8#discussion_r1073611789

It was moved to another issue, to allow us to merge the PR and unblock other stories depending on it.

**Comment by jimmykarily** [comment]
Fixed here: https://github.com/kairos-io/kcrypt-challenger/pull/10/files#diff-94d75f189ea8b05ff1476dce6652e33bf43bdb1fddade2736156f8366b4eb22bR93-R107
